File: springboot-backend/src/main/java/com/wedspot/backend/services/IAuthService.java
package com.wedspot.backend.services;

import com.wedspot.backend.Model.*;

public interface IAuthService {
    APIResponse login(LoginRequest request);
    APIResponse register(RegisterRequest request);
    APIResponse getAllUsers();
    APIResponse getUser(Long id);
    
    // Management & Security
    APIResponse forgotPassword(ForgotPasswordRequest request);
    APIResponse resetPassword(ResetPasswordRequest request);
    APIResponse verifyOtp(VerifyOtpRequest request);
    APIResponse verifyToken(String token);
    APIResponse logout(String token);
}

File: backend1/Services/IAuthService.cs
using backend.Models;

namespace backend.Services
{
    public interface IAuthService
    {
        Task<APIResponse> Login(LoginRequest request);
        Task<APIResponse> Register(RegisterRequest request);
        Task<APIResponse> ForgotPassword(ForgotPasswordRequest request);
        Task<APIResponse> ResetPassword(ResetPasswordRequest request);
        Task<APIResponse> VerifyOtp(VerifyOtpRequest request);
        Task<APIResponse> VerifyToken(string token);
        Task<APIResponse> Logout(string token);
    }
}
